packages/core/utils/src/dml/modifiers/nullable.ts

```diff
@@ -12,9 +15,11 @@ export class NullableModifier<T> {
    * The parent schema on which the nullable modifier is
    * applied
    */
-  #schema: SchemaType<T>
+  #schema: {
+    parse(fieldName: string): MaybeFieldMetadata
+  }
 
-  constructor(schema: SchemaType<T>) {
+  constructor(schema: { parse(fieldName: string): MaybeFieldMetadata }) {
```

Comment:
Q: should we use `Pick<SchemaType<T>, 'parse'>` instead? just to prevent divergeance?

Reply:
If we agree on that then there is other places as well

Reply:
The modifier could be used with both `RelationshipType` and `SchemaType`. Ideally it should care what the `parse` returns except `nullable` and `optional` fields.

So it feels natural to have a sub-type that represents a MaybeField. Be it relationship or a property

Reply:
ah yes indeed it can be used with both 👍